src/index.js

```diff
@@ -0,0 +1,158 @@
+    if (data) {
+      tasks = JSON.parse(data);
+      renderTasks();
+    }
+  };
+
+  // To-do list data
+  let tasks = [];
+
+  // Create the to-do list functionality
+  const todoContent = document.querySelector('.task-content');
+  const inputBox = document.getElementById('input-to-do');
+  const inputListContainer = document.getElementById('list-container');
+  const addButton = document.querySelector('.add-button');
+  const errorMessage = document.createElement('p');
+  errorMessage.classList.add('error-message');
+  errorMessage.style.display = 'none';
+  todoContent.appendChild(errorMessage);
+
+  // Function to render tasks on the list
+  const renderTasks = () => {
+    inputListContainer.innerHTML = '';
+    tasks.forEach((task, index) => {
+      const li = document.createElement('li');
+      li.textContent = task.description;
+      if (task.completed) {
+        li.classList.add('checked');
+      }
+      const deleteButton = document.createElement('span');
+      deleteButton.innerHTML = '\u2716';
+      deleteButton.addEventListener('click', () => {
+        deleteTask(index);
+      });
+      li.appendChild(deleteButton);
+      const checkbox = document.createElement('input');
+      checkbox.type = 'checkbox';
+      checkbox.checked = task.completed;
+      checkbox.addEventListener('change', (e) => {
+        updateTaskStatus(index, e.target.checked);
+      });
+      inputListContainer.appendChild(li);
+    });
+    saveData();
+  };
+
+  // Function to add a new task
+  const addTask = () => {
+    if (inputBox.value === '') {
+      errorMessage.textContent = 'Please enter a task!';
+      errorMessage.style.display = 'block';
+    } else {
+      const newTask = {
+        description: inputBox.value,
+        completed: false,
+        index: tasks.length + 1,
+      };
+      tasks.push(newTask);
+      inputBox.value = '';
+      errorMessage.style.display = 'none';
+      renderTasks();
+    }
+  };
+
+  // Function to delete a task
+  const deleteTask = (index) => {
+    tasks.splice(index, 1);
+    tasks.forEach((task, i) => {
+      task.index = i + 1;
+    });
+    renderTasks();
+  };
+
+  // Function to edit a task description
+  // eslint-disable-next-line no-unused-vars
+  const editTaskDescription = (index, newDescription) => {
+    tasks[index].description = newDescription;
+    renderTasks();
+  };
+
+  addButton.addEventListener('click', addTask);
+  inputBox.addEventListener('keydown', (event) => {
+    if (event.key === 'Enter') {
+      addTask();
+    }
+  });
+
+  inputListContainer.addEventListener('click', (e) => {
+    if (e.target.tagName === 'LI') {
+      const index = Array.from(inputListContainer.children).indexOf(e.target);
+      tasks[index].completed = !tasks[index].completed;
+      renderTasks();
+    }
+  });
+
+  const clearAllTasks = () => {
+    tasks = [];
+    renderTasks();
+  };
+
+  clearAll.addEventListener('click', clearAllTasks);
```

Comment:
- [ ] The clear all completed button suppose to clear all completed todos not all in your case you clear all todos, kindly consider fixing this.